Mage.Sets/src/mage/cards/m/MirkwoodElk.java

```diff
@@ -0,0 +1,87 @@
+package mage.cards.m;
+
+import java.util.UUID;
+import mage.MageInt;
+
+import mage.abilities.Ability;
+import mage.abilities.keyword.TrampleAbility;
+import mage.abilities.common.EntersBattlefieldOrAttacksSourceTriggeredAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.SubType;
+import mage.constants.Outcome;
+import mage.filter.FilterCard;
+import mage.target.common.TargetCardInYourGraveyard;
+
+import mage.game.Game;
+import mage.players.Player;
+import mage.constants.Zone;
+
+/**
+ * @author rullinoiz
+ */
+public final class MirkwoodElk extends CardImpl {
+
+    private static final FilterCard filter = new FilterCard("Elf card from your graveyard");
+
+    static {
+        filter.add(SubType.ELF.getPredicate());
+    }
+
+    public MirkwoodElk(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{5}{G}");
+        
+        this.subtype.add(SubType.ELK);
+        this.power = new MageInt(6);
+        this.toughness = new MageInt(6);
+
+        // Trample
+        this.addAbility(TrampleAbility.getInstance());
+
+        // Whenever Mirkwood Elk enters the battlefield or attacks, return target Elf card from your graveyard to your hand.
+        // You gain life equal to that card's power.
+        Ability ability = new EntersBattlefieldOrAttacksSourceTriggeredAbility( new MirkwoodElkEffect(), false );
+        ability.addTarget( new TargetCardInYourGraveyard(filter) );
+
+        this.addAbility(ability);
+    }
+
+    private MirkwoodElk(final MirkwoodElk card) {
+        super(card);
+    }
+
+    @Override
+    public MirkwoodElk copy() {
+        return new MirkwoodElk(this);
+    }
+}
+
+class MirkwoodElkEffect extends OneShotEffect {
+
+    public MirkwoodElkEffect() {
```

Comment:
nit, these constructors in the custom effect shouldn't be public (no access modifier needed)

Reply:
just wondering what the issue would be if the effect constructor had an access modifier or not so i can be aware next time. all the other effect constructors i've looked through are public, so i'm not sure what makes this one different

Reply:
Mostly it's annoying if the IDE suggests autocompleting custom effects in other areas of the project, increases the chance for a mistake of selecting the wrong effect.. It's a minor thing that is often overlooked, so it's all over the place in the codebase. It shouldn't cause any issue for the card implementation itself.

For general use effect/ability classes: typically public constructors, and protected copy constructor
For card-specific effect/ability classes: typically package-private constructors, and private copy constructor

Reply:
alright, i'll keep that in mind for next time